pkg/streamingpromql/optimize/ast/prune_toggles.go

```diff
@@ -16,13 +16,29 @@ import (
 // e.g. `(avg(rate(foo[1m]))) and on() (vector(1) == 1)` -> `(avg(rate(foo[1m])))`
 // e.g. `(avg(rate(foo[1m]))) and on() (vector(1) == -1)` -> `(vector(1) == -1)`
 type PruneToggles struct {
-	mapper astmapper.ASTMapper
+	mapper *ASTExprMapperWithState
+
+	pruneTogglesMetrics
+}
+
+type pruneTogglesMetrics struct {
+	pruneTogglesAttempts  prometheus.Counter
+	pruneTogglesSuccesses prometheus.Counter
```

Comment:
Given these fields are in a struct called `pruneTogglesMetrics`, the `pruneToggles` prefixes on both seem unnecessary to me.

Reply:
We can access them directly like `p.pruneTogglesAttempts.Inc()` without the struct `pruneTogglesMetrics`. If you think it's better should I change it to remove `pruneTogglesMetrics`?